x-pack/elastic-agent/pkg/agent/cmd/enroll.go

```diff
@@ -308,6 +311,24 @@ func enroll(streams *cli.IOStreams, cmd *cobra.Command, args []string) error {
 
 	ctx := handleSignal(context.Background())
 
+	if localFleetServer := fServer != ""; localFleetServer {
```

Comment:
Could you move this to a separate function and then have some unit tests on it to validate the behaviour?

Reply:
I think that port should be configurable. What if the user is already using that port for any other process running?

Reply:
This line is really hard to understand.  Can you break it up somehow?

Reply:
i made it configurable to make the option available for configuring port. if user wants to have it on some managed port. i can take it out if that's something not important atm